### PR TITLE
base

### PR DESCRIPTION
**Pull request by BelohvostovAlex**

### CODE REVIEW

File: src/components/Modal/Modal.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        <RoundButton
+          btnText="X"
+          btnClass="round-btn round-btn--gray round-btn--toRight"
+          onClickBtn={onClose}
+        />
+        <label htmlFor="modal-form__text">Title</label>
+        <input
+          type="text"
+          name="modal-from__text"
+          id="modal-form__text"
```
Comment (igorTT) [inline src/components/Modal/Modal.tsx:28]: Don't use bem notation in the id naming

File: src/components/Button/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+
+import './buttons.scss';
+import { ButtonProps } from './interfaces';
+
+export const Button: React.FC<ButtonProps> = ({
+  btnText,
+  btnClass,
+  onClickBtn,
```
Comment (igorTT) [inline src/components/Button/Button.tsx:9]: Replace with onClick

File: src/components/Button/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+
+import './buttons.scss';
+import { ButtonProps } from './interfaces';
+
+export const Button: React.FC<ButtonProps> = ({
+  btnText,
+  btnClass,
```
Comment (igorTT) [inline src/components/Button/Button.tsx:8]: replace with className

File: src/components/Button/Button.tsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+
+import './buttons.scss';
+import { ButtonProps } from './interfaces';
+
+export const Button: React.FC<ButtonProps> = ({
+  btnText,
```
Comment (igorTT) [inline src/components/Button/Button.tsx:7]: replace with text or title

File: src/components/Modal/Modal.tsx [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  onClose,
+  createTodo,
+}) => {
+  return (
+    <div className="modal">
+      <form className="modal-form">
+        <RoundButton
+          btnText="X"
+          btnClass="round-btn round-btn--gray round-btn--toRight"
+          onClickBtn={onClose}
```
Comment (igorTT) [inline src/components/Modal/Modal.tsx:22]: replace with 'text', 'className', 'onClick'

File: src/components/Modal/Modal.tsx [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+          cols={30}
+          rows={10}
+          value={description}
+          onChange={handleDescription}
+        ></textarea>
+        <div className="modal-form__buttons">
+          <Button
+            btnText="Create"
+            btnClass="button button--blue"
+            onClickBtn={() => createTodo(title, description)}
```
Comment (igorTT) [inline src/components/Modal/Modal.tsx:45]: The same as above
If you want it'll be a plain component- you should pass only the onSubmit handler. That won't get any arguments here. You have title and description in the upper level

File: src/components/Modal/ModalContainer.tsx [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+import { ModalContainerProps } from './interfaces';
+import { Modal } from './Modal';
+
+export const ModalContainer: React.FC<ModalContainerProps> = ({
+  onClose,
+  addTodo,
+}) => {
+  const [title, handleTitle] = useInput();
+  const [description, handleDescription] = useInput();
+  const createTodo = (title: string, description: string) => {
```
Comment (igorTT) [inline src/components/Modal/ModalContainer.tsx:14]: Add an empty line before function. Don't save the place on the screen

File: src/components/Modal/ModalContainer.tsx [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+    addTodo({ id: v4(), title, description });
+    onClose();
+  };
+  return (
```
Comment (igorTT) [inline src/components/Modal/ModalContainer.tsx:18]: Add empty line

File: src/components/Todo/Todo.tsx [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+import { RoundButton } from '../RoundButton/RoundButton';
+
+import './todo.scss';
+
+export const Todo: React.FC<TodoProps> = ({ id, title, descr, deleteTodo }) => {
+  return (
+    <div className="todo">
+      <div className="todo-text">
+        <h2 className="todo-text__title">{title}</h2>
+        <h3 className="todo-text__descr">{descr}</h3>
```
Comment (igorTT) [inline src/components/Todo/Todo.tsx:13]: desc -> description

File: src/pages/TodoPage/TodoPage.tsx [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+}) => {
+  return (
+    <div className="todo-page">
+      <PageHeader title="Todo list" />
+      <div className="container">
+        {!todos.length ? (
+          <EmptyBlock text="There are nothing here... Please create one" />
+        ) : (
+          <div className="todo-page__wrapper">
+            {todos.slice(-5).map(({ id, title, description }) => {
```
Comment (igorTT) [inline src/pages/TodoPage/TodoPage.tsx:26]: Slice the array in the upper level

File: src/pages/TodoPage/TodoPage.tsx [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+  deleteTodo,
+  addTodo,
+  visibleModal,
+  handleVisibleModal,
+}) => {
+  return (
+    <div className="todo-page">
+      <PageHeader title="Todo list" />
+      <div className="container">
+        {!todos.length ? (
```
Comment (igorTT) [inline src/pages/TodoPage/TodoPage.tsx:22]: Replace with 'hasTodo' boolean value

**Review by igorTT — CHANGES_REQUESTED** [review]
image
I don't see the 'sass' dependency in the project.
image
There is a warning in the terminal
The modal window width depends on the screen size. Replace this behavior with fixed size
The modal window inputs should use placeholders.
image
Should be in the top right corner
Todo card also depends on the screen size. Replace with a static size
Check styles one more time.

File: src/components/Modal/Modal.tsx (hunk 1 above)
Comment (BelohvostovAlex) [reply src/components/Modal/Modal.tsx:28]: changed

File: src/components/Button/Button.tsx (hunk 2 above)
Comment (BelohvostovAlex) [reply src/components/Button/Button.tsx:9]: done

File: src/components/Button/Button.tsx (hunk 3 above)
Comment (BelohvostovAlex) [reply src/components/Button/Button.tsx:8]: done

File: src/components/Button/Button.tsx (hunk 4 above)
Comment (BelohvostovAlex) [reply src/components/Button/Button.tsx:7]: done

File: src/components/Modal/Modal.tsx (hunk 5 above)
Comment (BelohvostovAlex) [reply src/components/Modal/Modal.tsx:22]: done

File: src/components/Modal/ModalContainer.tsx (hunk 7 above)
Comment (BelohvostovAlex) [reply src/components/Modal/ModalContainer.tsx:14]: done

File: src/components/Modal/ModalContainer.tsx (hunk 8 above)
Comment (BelohvostovAlex) [reply src/components/Modal/ModalContainer.tsx:18]: done

File: src/components/Todo/Todo.tsx (hunk 9 above)
Comment (BelohvostovAlex) [reply src/components/Todo/Todo.tsx:13]: done

File: src/pages/TodoPage/TodoPage.tsx (hunk 10 above)
Comment (BelohvostovAlex) [reply src/pages/TodoPage/TodoPage.tsx:26]: done

File: src/components/Modal/Modal.tsx (hunk 6 above)
Comment (BelohvostovAlex) [reply src/components/Modal/Modal.tsx:45]: done

File: src/pages/TodoPage/TodoPage.tsx (hunk 11 above)
Comment (BelohvostovAlex) [reply src/pages/TodoPage/TodoPage.tsx:22]: done

**Review by igorTT — CHANGES_REQUESTED** [review]

File: src/components/Todo/Todo.tsx [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+    <div className="todo">
+      <div className="todo-text">
+        <h2 className="todo-text__title">{title}</h2>
+        <h3 className="todo-text__descr">{description}</h3>
+      </div>
+      <div className="todo-buttons">
+        <RoundButton
+          text="X"
+          className="round-btn round-btn--red"
+          onClick={() => deleteTodo(id)}
```
[the remark this answers is not in the document]
Comment (BelohvostovAlex) [reply src/components/Todo/Todo.tsx:24]: done

**Review by igorTT — APPROVED** [review]

**Pull request merged**